### ISSUE TITLE
Seeing quite a few MAC mismatch errors

### ISSUE THREAD

**Issue by myleshorton** [Yawning/obfs4#28]
We're seeing thing fairly randomly on a subset of obfs4 client and servers all with the same basic configuration. Possibly some sort of race condition or any issue on startup?

It's always possible there's some government in the middle messing with the traffic, but it seems unlikely.

```
mismatch: Dervied: 603f664cca9202933a29ea7f6ad18a87 Received: 7ce5ffcf995cde6c1580836c2f21eb15.
```

**Comment by Yawning** [comment]
The primary cause I've seen of this that isn't operator error is when the clock skew between the client and server is too large.

Due to a bug (fixed in 62057625eaba2a555967b9615039d43dda071360) the clocks must agree exactly about the number of hours since the unix epoch.  With the fix, this limit is relaxed to allow skew of +- 1 hour.

**Comment by Yawning** [comment]
Closing as the fix apparently did the trick per separate correspondence.
